content/questions/Recursion with slice function/index.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Recursion with slice()
+tags:
+- recursion
+- slice
+order: 17
+date: '2019-10-05'
+answers:
+- 'Hello World'
+- 'e'
+- 'H // correct'
+- 'd'
+---
+
+Consider the following recursive function. If we pass the string `"Hello World"` to it, what gets logged?
+
+```javascript
+var speak = str => {
```

Comment:
Can you remove extra line breaks and add indenting to this function?

I think it should be something like:

```javascript
var speak = str => {
  if (str.length > 1) {
    return speak(str.slice(0, -1));
  }
  return str;
}
```

Reply:
Extra line break removed. ✅

Reply:
![RunJS Mon Oct 07 2019 14:09:25 GMT+0530 (India Standard Time)](https://user-images.githubusercontent.com/45743584/66296915-39cb1280-e90c-11e9-9704-0f689ae207ee.png)
return speak(str.slice(0,1));